asf_heat_pump_suitability/analysis/hn_zones/plot_comparison_of_hn_zones.py

```diff
@@ -0,0 +1,225 @@
+        output_dir (str): Output directory to save the plot image.
+    """
+    gdf = pyogrio.read_dataframe(
+        liverpool_heat_network_zones_filepath, layer="heat-network-zone-map-Liverpool"
+    )
+    fig, ax = plt.subplots(figsize=(12, 6))
+    gdf.plot(ax=ax, color="blue", alpha=0.5)
+    liverpool_hp_suitability_gdf[
+        liverpool_hp_suitability_gdf["DESNZ_pilot_score"] == 1
+    ].plot(ax=ax, color="pink", alpha=0.5)
+    plt.title("Overlay of DESNZ Pilot Score = 1 on Liverpool Heat Network Zones")
+    plt.axis("off")
+    plt.savefig(os.path.join(output_dir, "overlay_desnz_pilot.png"))
+    plt.show()
+
+
+def plot_absolute_error_map(
+    hp_suitability_gdf: gpd.GeoDataFrame,
+    score: int,
+    title: str,
+    filename: str,
+    output_dir: str = OUTPUT_DIR,
+):
+    """
+    Plot an absolute error map for areas in Liverpool based on the DESNZ pilot score,
+    overlaying regions with a different score as boundaries.
+
+    Args:
+        df (gpd.GeoDataFrame): GeoDataFrame containing Liverpool's LSOA geometries and
+                               heat pump suitability scores, including absolute error values.
+        score (int): DESNZ pilot score value for which to highlight areas.
+                     Only regions matching this score will be color-filled.
+        title (str): Title for the plot.
+        filename (str): Path to save the plot image as a PNG file.
+        output_dir (str): Output directory to save the plot image.
+
+    This function:
+        - Plots regions where DESNZ pilot score matches the specified `score`,
+          coloring them by 'absolute_error' values.
+        - Overlays boundaries of areas where DESNZ pilot score does not match the specified `score`.
+        - Displays a legend and saves the plot as a PNG file.
+    """
+    fig, ax = plt.subplots(figsize=(12, 6))
+    hp_suitability_gdf[hp_suitability_gdf["DESNZ_pilot_score"] == score].plot(
+        column="absolute_error", ax=ax, legend=True, cmap="inferno"
+    )
+    hp_suitability_gdf[hp_suitability_gdf["DESNZ_pilot_score"] != score].plot(
+        ax=ax, edgecolor="black", facecolor="none", linewidth=0.5
+    )
+    plt.title(title)
+    plt.axis("off")
+    plt.savefig(os.path.join(output_dir, filename))
+    plt.show()
+
+
+def main():
+    """
+    Main function to load, preprocess, and visualise data.
+    """
+    liverpool_hp_suitability_scores_pd, liverpool_hp_suitability_lsoas = load_data()
+    liverpool_lsoa_geometries_gdf = preprocess_data(liverpool_hp_suitability_lsoas)
+    plot_lsoa_geometries(liverpool_lsoa_geometries_gdf)
+    liverpool_hp_suitability_gdf = merge_data(
+        liverpool_hp_suitability_scores_pd, liverpool_lsoa_geometries_gdf
+    )
+    plot_overlay(liverpool_hp_suitability_gdf)
+    plot_absolute_error_map(
+        liverpool_hp_suitability_gdf,
+        score=1,
+        title="Absolute Error Map for DESNZ Pilot Score = 1 for Liverpool",
```

Comment:
I think this map and below need a slightly more descriptive plot title / legend to be clear what the error is and what DESNZ pilot score means.